document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/parser/MongoFilterTypeExpressionParser.java

```diff
@@ -7,27 +7,41 @@
 import java.util.Optional;
 import java.util.function.Function;
 import org.hypertrace.core.documentstore.Key;
+import org.hypertrace.core.documentstore.expression.impl.ArrayRelationalFilterExpression;
+import org.hypertrace.core.documentstore.expression.impl.DocumentArrayFilterExpression;
 import org.hypertrace.core.documentstore.expression.impl.KeyExpression;
 import org.hypertrace.core.documentstore.expression.impl.LogicalExpression;
 import org.hypertrace.core.documentstore.expression.impl.RelationalExpression;
 import org.hypertrace.core.documentstore.expression.type.FilterTypeExpression;
+import org.hypertrace.core.documentstore.mongo.query.parser.filter.MongoRelationalFilterParserFactory.MongoRelationalFilterContext;
 import org.hypertrace.core.documentstore.parser.FilterTypeExpressionVisitor;
 import org.hypertrace.core.documentstore.query.Query;
 
 public final class MongoFilterTypeExpressionParser implements FilterTypeExpressionVisitor {
 
   private static final String FILTER_CLAUSE = "$match";
 
+  private final MongoRelationalFilterContext relationalFilterContext;
+
+  public MongoFilterTypeExpressionParser() {
+    this(MongoRelationalFilterContext.DEFAULT_INSTANCE);
+  }
+
+  public MongoFilterTypeExpressionParser(
+      final MongoRelationalFilterContext relationalFilterContext) {
+    this.relationalFilterContext = relationalFilterContext;
+  }
+
   @SuppressWarnings("unchecked")
   @Override
   public Map<String, Object> visit(final LogicalExpression expression) {
-    return new MongoLogicalExpressionParser().parse(expression);
+    return new MongoLogicalExpressionParser(relationalFilterContext).parse(expression);
   }
 
   @SuppressWarnings("unchecked")
   @Override
   public Map<String, Object> visit(final RelationalExpression expression) {
-    return new MongoRelationalExpressionParser().parse(expression);
+    return new MongoRelationalExpressionParser().parse(expression, relationalFilterContext);
```

Comment:
Any specific reason to parse `context` as part of method `parse`. But on the above, the context is passed as part of constructor - `MongoLogicalExpressionParser`. If, the object is supposed to run always in the context, we can pass it in constructor.

Reply:
Makes sense. Sure, it can be passed in the constructor. I'll update it.